api/session.go

```diff
@@ -146,6 +146,7 @@ func (s *Session) Renew(id string, q *WriteOptions) (*SessionEntry, *WriteMeta,
 // session until a doneCh is closed. This is meant to be used in a long running
 // goroutine to ensure a session stays valid.
```

Comment:
I think we should add a clarifying comment here that if the context in WriteOptions has a deadline, then after the deadline has passed this will stop renewing the session.